Rubberduck.RegexAssistant/Atom.cs

```diff
@@ -0,0 +1,284 @@
+            {
+                if (specifier.Value.Contains("\\"))
+                {
+                    if (specifier.Value.EndsWith("-\\"))
+                    {
+                        // BOOM!
+                        throw new ArgumentException("Character Ranges that have incorrectly escaped characters as target are not allowed");
+                    }
+                    else if (specifier.Value.Length == 1)
+                    {
+                        // fun... we somehow got to grab a single backslash. Pattern is probably broken
+                        // alas for simplicity we just skip the incorrect backslash
+                        // TODO: make a warning from this.. how?? no idea
+                        continue;
+                    }
+                }
+                CharacterSpecifiers.Add(specifier.Value);
+            }
+        }
+
+        public string Description
+        {
+            get
+            {
+                return string.Format(InverseMatching 
+                    ? AssistantResources.AtomDescription_CharacterClass_Inverted 
+                    : AssistantResources.AtomDescription_CharacterClass
+                    , HumanReadableClass());
+            }
+        }
+
+        private string HumanReadableClass()
+        {
+            return string.Join(", ", CharacterSpecifiers); // join last with and?
+        }
+
+        public bool TryMatch(ref string text)
+        {
+            throw new NotImplementedException();
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj is CharacterClass)
+            {
+                return (obj as CharacterClass).specifier.Equals(specifier);
+            }
+            return false;
+        }
+    }
+
+    class Group : Atom
+    {
+        public static readonly string Pattern = @"(?<!\\)\((?<expression>.*)(?<!\\)\)";
+        private static readonly Regex Matcher = new Regex("^" + Pattern + "$");
+
+        private readonly IRegularExpression subexpression;
+        private readonly string specifier;
+
+        public Group(string specifier) {
+            Match m = Matcher.Match(specifier);
+            if (!m.Success)
+            {
+                throw new ArgumentException("The given specifier does not denote a Group");
+            }
+            subexpression = RegularExpression.Parse(m.Groups["expression"].Value);
+            this.specifier = specifier;
+        }
+
+        public string Specifier
+        {
+            get
+            {
+                return specifier;
+            }
+        }
+
+        public string Description
+        {
+            get
+            {
+                return string.Format(AssistantResources.AtomDescription_Group, specifier) + "\r\n" + subexpression.Description;
+            }
+        }
+
+        public bool TryMatch(ref string text)
+        {
+            throw new NotImplementedException();
+        }
+
+        public override bool Equals(object obj)
```

Comment:
It does fire up CA2218 ... I just ignored it for now. for one because I hate writing HashCode myself and VS didn't suggest an auto-generation and secondly because at this stage HashCode was not necessary yet